onelogin/get.go

```diff
@@ -107,30 +109,80 @@ func Get(app, provider string) (*awsprovider.Credentials, error) {
 		fmt.Scanln(&selection)
 
 		deviceID = fmt.Sprintf("%v", devices[selection-1].DeviceId)
+		deviceType = devices[selection-1].DeviceType
+
 	} else {
 		deviceID = fmt.Sprintf("%v", devices[0].DeviceId)
+		deviceType = devices[0].DeviceType
 	}
 
-	fmt.Print("Please enter the OTP from your MFA device: ")
-	var otp string
-	fmt.Scanln(&otp)
+	var samlAssertion string
+
+	var pushSupported = false
+
+	if deviceType == "OneLogin Protect" {
+		// push is supported by the selected OTP device
+		// if other are supporting push they should be added here.
+		pushSupported = true
+		s.Start()
```

Comment:
There are currently visual problems in the terminal because the spinner is started and stopped at the wrong places.
Following is the correct behavior:

```go
s.Start()
rMfa, err := c.VerifyFactor(token, &pMfa)
if err != nil {
    return nil, err
}
s.Stop()

pMfa.DoNotNotify = true

timeout := 30
interval := 4
fmt.Println(rMfa.Status.Message)
s.Start()
for rMfa.Status.Type == "pending" && timeout > 0 {
    time.Sleep(time.Duration(interval) * time.Second)
    rMfa, err = c.VerifyFactor(token, &pMfa)
    if err != nil {
        s.Stop()
        return nil, err
    }

    timeout -= interval
}
s.Stop()

if rMfa.Status.Type == "pending" {
    fmt.Println("Push timed out, please enter OTP token.")
    pushSupported = false
} else {
    samlAssertion = rMfa.Data
}
```

Reply:
It was actually my intention to have the spinner running while outputting the status message. That's why I added a space at the beginning and `\r` at end of the status output. I mean I've never seen another message than the one displayed, but the intention was to display a possibly changing message and I didn't want to have multiple lines of the same message.

Reply:
Yeah, I figured this is what you wanted.
The problem is - it creates visual problems as it is now. I prefer to have something a bit less fancy but stable.
And in any case, the spinner should start right before the long-running piece of code and stop as soon as possible.